server/aesdsocket.c

```diff
@@ -1,242 +1,344 @@
+                }
+		/*set the working directory to the root directory*/
+                if (chdir("/") == -1) {
+                   
+                    exit(-1);
+                }
+                /*redirect fd's 0,1,2 to /dev/null*/
+                open("/dev/null", O_RDWR); /*stdin*/
+                dup(0); 	/*stdout*/
+                dup(0);
+                printf("Daemon created\n");
+               
+            } 
             else 
             {
-                exit (0);
+                exit(0);
             }
         }
     }
 
+    //printf("opening /var/tmp/aesdsocketdata\n");
     
-    int listenfd;
-    listenfd = listen(sockfd, 5);
-    printf("Listening successful\n");
-
-    if (listenfd == -1)
-    {
-        perror("Error in Listening process \n");
-        return -1;
-    }
-
-    
-    /*file_fd = open("/var/tmp/aesdsocketdata",O_RDWR | O_CREAT | O_TRUNC, S_IRUSR | S_IWUSR | S_IRGRP |S_IWGRP | S_IROTH | S_IWOTH, 0777);*/
+    // open
     file_fd = open("/var/tmp/aesdsocketdata",O_RDWR | O_CREAT | O_TRUNC, 0777);
-    
-    if (file_fd == -1)
-    {
+
+    if (file_fd == -1) {
         perror("Error in opening file descriptor.");
     }
+   
     
-    int file_size = 0;
-    
-    char receive_buffer[BUF_SIZE];
-    
-    while(1)
-    {
-        client_len = sizeof(clientadd);
-        clientfd = accept(sockfd, (struct sockaddr *)&clientadd, &client_len);
-        if (clientfd == -1)
-        {
-            perror("error accepting\n");
-            return -1;
-        }
-        printf("Accepted connection from %s\n",inet_ntoa(clientadd.sin_addr));
-        syslog(LOG_DEBUG,"Accepted connection from %s\n",inet_ntoa(clientadd.sin_addr));  
+   
+    pthread_mutex_init(&lock, NULL);
 
-        bool status = false;
-        int buffer = BUF_SIZE;
-        int buffer_size = 0;
-        char *write_buf = malloc(sizeof(char) * buffer);
+    TAILQ_INIT(&head);
 
-        while(!status)
-        {
-            // receive the data from client
-            rec = recv(clientfd, receive_buffer, BUF_SIZE, 0);
-            if (rec == 0)
-            {
-                status = true;
-            } 
-            for (int i = 0; i < rec; i++)
-            {
-                
-                // appneding new line
- 		if (receive_buffer[i] == '\n')
-                {
-                    status = true;
-                }
-            }
-            if ((buffer - buffer_size) < rec)
-            {
-                buffer = buffer +rec;
-                write_buf = (char *) realloc(write_buf, sizeof(char) * buffer);
-            }
-            memcpy(write_buf + buffer_size, receive_buffer, rec);
-            buffer_size += rec;
+    int listenfd;
+    bool alarm_flag = false;
+
+    // listen
+    listenfd = listen(sockfd, 5);
+     if (listenfd == -1) 
+     {
+        perror("Error in Listening \n");
+       
+    }
+   
+    while (1) {
+        thread_data_t *data = (thread_data_t *) malloc(sizeof(thread_data_t));
```

Comment:
@swapnilghonge what will happen if a `TERM` signal occurs between the malloc at this line and the place where you add to the linked list in line 318?  I think you will leak the memory associated with `data`